meteor/client/ui/Settings/Studio/Routings.tsx

```diff
@@ -12,757 +10,1031 @@ import {
+									/>
+									<TextInputControl
+										modifiedClassName="bghl"
+										classNames="input text-input input-l"
+										value={value}
+										handleUpdate={handleUpdate}
+									/>
+								</div>
+							)}
+						</LabelAndOverrides>
+
+						{/** TODO: this needs the same checkbox to enable/disable as above */}
+						<DeviceMappingSettings
+							translationNamespaces={translationNamespaces}
+							mappedLayer={mappedLayer}
+							manifest={routeMappingSchema}
+							overrideHelper={tableOverrideHelper}
+							route={route}
+						/>
+					</>
+				) : null}
+			</div>
+		</div>
+	)
+}
 
 interface IDeviceMappingSettingsProps {
 	translationNamespaces: string[]
-	studio: DBStudio
-	attribute: string
 	manifest: MappingsSettingsManifest | undefined
 	mappedLayer: ReadonlyDeep<MappingExt> | undefined
+	overrideHelper: OverrideOpHelperArrayTable
+	route: WrappedOverridableItemNormal<RouteMapping>
 }
 
 function DeviceMappingSettings({
 	translationNamespaces,
-	attribute,
 	manifest,
-	studio,
 	mappedLayer,
-}: IDeviceMappingSettingsProps) {
-	const routeRemapping = objectPathGet(studio, attribute)
+	overrideHelper,
+	route,
+}: Readonly<IDeviceMappingSettingsProps>) {
+	const mappingType = route.computed?.remapping?.options?.mappingType ?? mappedLayer?.options?.mappingType
+	const mappingSchema = mappingType ? manifest?.mappingsSchema?.[mappingType] : undefined
 
-	const mappingType = routeRemapping?.mappingType ?? mappedLayer?.options?.mappingType
-	const mappingSchema = manifest?.mappingsSchema?.[mappingType]
-
-	if (mappingSchema && routeRemapping) {
+	if (mappingSchema) {
 		return (
-			<SchemaFormForCollection
+			<SchemaFormWithOverrides
 				schema={mappingSchema}
-				object={routeRemapping}
-				basePath={attribute}
 				translationNamespaces={translationNamespaces}
-				collection={Studios}
-				objectId={studio._id}
-				partialOverridesForObject={mappedLayer}
+				attr={'remapping.options'}
+				item={route}
+				overrideHelper={overrideHelper}
+				isRequired
 			/>
 		)
 	} else {
 		return null
 	}
 }
+
+interface IRenderExclusivityGroupsProps {
+	studio: DBStudio
+	toggleExpanded: (exclusivityGroupId: string, force?: boolean) => void
+	isExpanded: (exclusivityGroupId: string) => boolean
+	routeSetsFromOverrides: WrappedOverridableItem<StudioRouteSet>[]
+	exclusivityGroupsFromOverrides: WrappedOverridableItem<StudioRouteSetExclusivityGroup>[]
+}
+
+function RenderExclusivityGroups({
+	studio,
+	toggleExpanded,
+	isExpanded,
+	routeSetsFromOverrides,
+	exclusivityGroupsFromOverrides,
+}: Readonly<IRenderExclusivityGroupsProps>): React.JSX.Element {
+	const { t } = useTranslation()
+
+	const saveExclusivityOverrides = React.useCallback(
+		(newOps: SomeObjectOverrideOp[]) => {
+			Studios.update(studio._id, {
+				$set: {
+					'routeSetExclusivityGroupsWithOverrides.overrides': newOps,
+				},
+			})
+		},
+		[studio._id, studio.routeSetExclusivityGroupsWithOverrides]
```

Comment:
I don't see a reason for this dependency on `studio.routeSetExclusivityGroupsWithOverrides`. 
Having that dependency, means that each time the `studio.routeSetExclusivityGroupsWithOverrides` property changes (ie, following each change made by the user), this callback will be recreated, which will in turn cause `exclusivityOverrideHelper` to be recreated and will propogate re-renders to everywhere which has a reference to that 
```suggestion
		[studio._id]
```